src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml

```diff
@@ -80,6 +80,9 @@
         </service>
 
         <service id="router" alias="router.default" />
+        <service id="Symfony\Component\Routing\RouterInterface" alias="router.default" public="false" />
```

Comment:
the alias should point to ``router``, not to ``router.default``, to inject the right service when overwriting or decorating the router